### DISCUSSION TITLE
What is the proper way to load the project into Intellij?

### DISCUSSION THREAD

**Discussion by starksm64** [eclipse-ee4j/glassfish#25200] [Q&A] [answered]
## Environment Details
* GlassFish Version (and build number): github 8.0-JDK17 branch
* JDK version: 17
* OS: mac os
* Database: *

----------

## Problem Description
Every time I brows the glassfish project in Intellij there are problems with navigating the code. It seems like I am not loading the correct top level pom. I'm just opening the glassfish repo with the 8.0-JDK17 branch checked out in Intellij, so it is just loading the room pom.xml. I have enabled the snapshot profile, but if I navigate to the appserver/web/weld-integration/src/main/java/org/glassfish/cdi/hk2/HK2IntegrationExtension.java class, go to the `injectionTargetObserver(@Observes ProcessInjectionTarget<T> pit)` method, and try to find the declarations or usage of the convertInjectionPointToInjectee(...) method call on line 70, nothing is returned. 

I can brute force search the codebase for this method and find it in:
appserver/web/weld-integration/src/main/java/org/glassfish/cdi/hk2/HK2IntegrationUtilities.java

but this slows down inspecting the codebase.

There must be some pom.xml file that needs to be loaded.

## Impact of Issue
Makes it difficult to try to contribute to the project.

**Comment by starksm64** [comment, marked as answer]
@arjantijms indicated that he loads the appserver/featuresets/debug module in Eclipse IDE, and doing the same from Intellij appears to have fixed the missing dependencies.

**Comment by dmatej** [comment]
I assume that the problem is that GF uses custom Maven lifecycle and IDEs are not prepared for that too well. However Eclipse is getting better, it works well for me, just sometimes it offers classes from glassfish-embedded.jar which don't have sources as it is basically a repackaged jar.
So Arjan created the "debug" module which partially offers a workaround for this.
I don't need it probably just because I always import all modules in the tree - once some dependency link breaks, it causes issues like this.
